src/main/java/edu/wsu/SnakePane.java

```diff
@@ -1,53 +1,146 @@
-  private Color bodyColor = Color.LIMEGREEN;
-  private Color headColor = Color.RED;
-  private String name = "Nick";
-
-//  public SnakePane(int l, Color bc, Color hc, String n){
-//    bodyLength = l;
-//    bodyColor = bc;
-//    headColor = hc;
-//    name = n;
-//  }
+
+  private int radius = 10;
+  private int width; //window width
+  private int height; //window height
+  private int direction = 2; //0 = up, 1 = down, 2 = right, 3 = left
+  private int bodyLength;
+  private int score;
+  private int numFruits;
+  private Color bodyColor;
+  private Color headColor;
+  private Color fruitColor;
+  private String name;
+
+  private AnimationTimer time;
+  private List<Circle> snake;
+  private List<Circle> fruits;
+
+  public SnakePane(){
+    width = 600;
+    height = 400;
+    bodyLength = 10;
+    score = 0;
+    numFruits = 2;
+    bodyColor = Color.LIMEGREEN;
+    headColor = Color.RED;
+    fruitColor = Color.RED;
+    name = "Snake";
+    snake = new LinkedList<>();
+  }
+
+  public SnakePane(int width, int height, int bodylength, int numFruits, Color bodyColor, Color headColor, Color fruitColor, String name){
+    this.width = width * 100;
+    this.height = height * 100;
+    this.bodyLength = bodylength;
+    score = 0;
+    this.numFruits = numFruits;
+    this.bodyColor = bodyColor;
+    this.headColor = headColor;
+    this.fruitColor = fruitColor;
+    this.name = name;
+    snake = new LinkedList<>();
+  }
 
   private List<Circle> createSnake(){
-    List<Circle> snake = new LinkedList<>();
+    List<Circle> body = new LinkedList<>();
 
-    int radius = 10;
+    var head = new Circle((width - (radius * 2))/2,(height - (radius * 2))/2, radius, headColor);
+    body.add(head);
+
+    for(int i = 1; i <= bodyLength; i++){
+      Circle bodySegment = new Circle((width - (radius * 2))/2 - (i * radius * 2), (height - (radius * 2))/2,radius,bodyColor);
+      body.add(bodySegment);
+    }
+
+    return body;
+  }
 
-    var head = new Circle(300,200, radius, headColor);
-    snake.add(head);
+  private void drawSnake(List<Circle> snake, Pane pane){ pane.getChildren().addAll(snake); }
 
-    int bodyLength = 10;
-    for(int i = 1; i <=bodyLength; i++){
-      Circle bodySegment = new Circle(300 - (i * radius * 2), 200,radius,bodyColor);
-      snake.add(bodySegment);
+  private List<Circle> createFruits(){
+    List <Circle> fruits = new LinkedList<>();
+    while(fruits.size() < numFruits){
+      Circle fruit = createFruit();
+      if(fruits.size() != 0) {
+        for (int i = 0; i < fruits.size(); i++) {
+          if(fruit.getCenterY() == fruits.get(i).getCenterY() && fruit.getCenterX() == fruits.get(i).getCenterX()){
+            continue;
+          }
+        }
+      }
+      fruits.add(fruit);
     }
-    return snake;
+    return fruits;
+  }
+
+  private Circle createFruit(){
+    Circle fruit;
+    Random location = new Random();
+    int x = location.nextInt(width/(2 * radius)) * (2 * radius) + radius;
```

Comment:
Good - this will force the fruit to be located in coordinates that are multiple of 2*radius -- making collision detection trivial.